llama-cpp-2/src/sampling.rs

```diff
@@ -213,8 +181,8 @@ impl LlamaSampler {
         penalize_nl: bool,
         ignore_eos: bool,
```

Comment:
nit: these may be worth turning into enums. Call sites like 

```rust
Self::penalties(
            model.n_vocab(),
            model.token_eos().0,
            model.token_nl().0,
            penalty_last_n,
            penalty_repeat,
            penalty_freq,
            penalty_present,
            false,
            true,
        )
```

are hard to figure out what true and false mean

Reply:
I expect this method to be used very rarely compared to `penalties_simple`, so I'm not sure that creating two public enums is worth it for this.